### PR TITLE
 Improved error handling and parsing in parsing_utils.py

### PR DESCRIPTION
**Pull request by stefanMadzharov**
Please check the type of change your PR introduces:

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

Issue Number: [230](https://github.com/keep-starknet-strange/garaga/issues)

# Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: hydra/garaga/starknet/groth16_contract_generator/parsing_utils.py
Comment (feltroidprime) [inline hydra/garaga/starknet/groth16_contract_generator/parsing_utils.py:1]: Should replace ValueError by KeyPatternNotFound on line 210

**Review by feltroidprime — CHANGES_REQUESTED** [review]

File: hydra/garaga/starknet/groth16_contract_generator/parsing_utils.py [hunk 1]
```diff
@@ -18,9 +18,9 @@
 from garaga.precompiled_circuits.multi_miller_loop import MultiMillerLoopCircuit
 
 # https://github.com/risc0/risc0-ethereum/blob/main/contracts/src/groth16/ControlID.sol
-RISC0_CONTROL_ROOT = 0x8B6DCF11D463AC455361B41FB3ED053FEBB817491BDEA00FDB340E45013B852E
+RISC0_CONTROL_ROOT = 0xA516A057C9FBF5629106300934D48E0E775D4230E41E503347CAD96FCBDE7E2E
 RISC0_BN254_CONTROL_ID = (
-    0x05A022E1DB38457FB510BC347B30EB8F8CF3EDA95587653D0EAC19E1F10D164E
+    0x0EB6FEBCF06C5DF079111BE116F79BD8C7E85DC9448776EF9A59AAF2624AB551
 )
```
Comment (feltroidprime) [inline hydra/garaga/starknet/groth16_contract_generator/parsing_utils.py:25]: This shouldn't change. What happened ?

**Review by feltroidprime — CHANGES_REQUESTED** [review]

File: hydra/garaga/starknet/groth16_contract_generator/parsing_utils.py (hunk 1 above)
Comment (stefanMadzharov) [reply hydra/garaga/starknet/groth16_contract_generator/parsing_utils.py:25]: I was debugging my program and didn't see that this it was included in the commit.

Comment (stefanMadzharov) [reply hydra/garaga/starknet/groth16_contract_generator/parsing_utils.py]: Also, I am not sure they are the latest values. The latest values should be those [here](https://docs.rs/risc0-circuit-recursion/1.1.2/src/risc0_circuit_recursion/control_id.rs.html#15-180). You can check [Issue#234](https://github.com/keep-starknet-strange/garaga/issues/234) I have explained my train of thoughts there.

**Comment by feltroidprime** [comment]
Lgtm

**Pull request merged**